src/Injector/AbstractInjector.php

```diff
@@ -162,6 +181,155 @@ public function inject($package, $type, IOInterface $io)
     }
 
     /**
+     * Inject component $package after all dependencies into $config and return true.
+     * If any of dependencies is not registered the method will write error to $io
+     * and will also return true, to prevent injecting this package later.
+     * Method return false only in case when dependencies for the package are not found.
+     *
+     * @param string $package
+     * @param string $config
+     * @param IOInterface $io
+     * @return bool
+     */
+    private function injectAfterDependencies($package, $config, IOInterface $io)
+    {
+        $moduleClass = sprintf('%s\\%s', $package, 'Module');
+        if (class_exists($moduleClass) && method_exists($moduleClass, 'getModuleDependencies')) {
+            $module = new $moduleClass();
+            $dependencies = $module->getModuleDependencies();
+
+            if ($dependencies) {
+                foreach ($dependencies as $dependency) {
+                    if (! $this->isRegisteredInConfig($dependency, $config)) {
+                        $io->write(sprintf(
+                            '<error>    Dependency %s is not registered in the configuration</error>',
+                            $dependency
+                        ));
+
+                        return true;
```

Comment:
I agree. However, because it runs post-package-install, we cannot block installation at this point (too late in the workflow).

Perhaps we should emit additional error information here: indicate that a dependency is missing, which may prevent the application from running, and maybe also indicate that after installing the dependency, the user should manually edit the configuration to ensure the dependency is listed *before* the package just installed.